ksy_style_guide.adoc

```diff
@@ -249,28 +248,36 @@ guesswork.
 
 * For simple non-repeated fields, use a simple singular form —
   i.e. `width`, `header`, `transaction_id`, `file`.
-* For repeated files (i.e. with `repeat: something`), use plular form
+* For a sequence (i.e. with `repeat: something`), use plular form
   — i.e. `files`, `transactions`.
 * Don't be overly verbose: use commonly understood abbreviations
   liberally, if it will improve readablity — i.e. `src_mac` or
   `src_mac_addr` instead of `source_media_access_control_address`
+* Repeated fileds which cannot be packed into a sequence should
+  have `id`s containing a number in the end. Numbers may have a
+  visually-obvious structure, like "the first digit is major, the second
+  one is minor".
 * For fields that are designed to be used to detect file type (AKA
-  "magic values"), use `magic` name, or, if there are several of them,
-  `magic1`, `magic2`, etc.
+  "magic values"), use `signature` or `magic` name, or, if there are
+  several of them, like `signature` or `magic0`, `magic1`, etc.
 * For reserved fields which are *known* to be unused, use `reserved`
-  name (or `reserved1`, `reserved2`, etc, if there are many of them)
+  name (or `reserved0`, `reserved1`, etc, if there are many of them)
 * For fields that designate *number / count* of something (in
-  particular, number of repetitions of some other structure), use
-  `num_` prefix and plular form — i.e. `num_questions`, `num_blocks`,
-  `num_nodes`
+  particular, number of repetitions of some other structure), use either
+  `_count` suffix or `count_of_` (what sounds better in your opinion)
+  prefix and a plular form — i.e. `count_of_questions`, `blocks_count`,
+  `nodes_count`
 * For fields that designate *offset* to some particular data structure,
-  use `ofs_` prefix and name of that data structure (as it would appear
-  in the file) — i.e. `ofs_block`, `ofs_queries`, `ofs_path`
+  use `_offset` suffix and name of that data structure (as it would
+  appear in the file) — i.e. `block_offset`, `queries_offset`, `path_offset`
 * For fields that designate *size* of some particular data structure
-  (in bytes or some other fixed units), use `len_` prefix and name of
-  that data structure — i.e. `len_block` (length of a single `block`
-  entry), `len_blocks` (total length of whole `blocks` array, made of
+  (in bytes or some other fixed units), use `_size` suffix and name of
+  that data structure — i.e. `block_size` (length of a single `block`
+  entry), `blocks_size` (total length of whole `blocks` array, made of
   `block` entries).
+* Fields of unknown/undetermined purpose, i.e. unfinished reverse
+  engineering work SHOULD either NOT TO HAVE `id` or HAVE an
+  `id` matching the `/unkn(?:own)?(_\w+)?\d*/` regular expression.
```

Comment:
This is, again, not how style guide should be written. It's pointless to leave so much choices to do for ksy developer. It's better to decide on this once and for all, and it will be easy for everyone — to remember, to follow, to understand, to write automated tools for that.

Reply:
1 Having an ```id``` for a property of unknown meaning is useful when doing reverse-engineering, especially if you have more than one such a property in the same struct.
2 But if a developer doesn't need that id, why to force him to have it?